### PR TITLE
[NON-MODULAR] [CONFIG] Prisoners now scale based on the security population. Defaults to 2 Prisoners for every Security person, maximum of 12.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

Prisoners now scale their available positions based on how many Security positions are filled.
Scaling defaults to 2 prisoners per 1 security member if no config is supplied. Prisoners via scaling are capped at 12 if no config is supplied.
Setting both configs to 0 disables this system.
Prisoners are now rolled after all other jobs for this purpose.

Closes #14081 

## How This Contributes To The Skyrat Roleplay Experience

See the conversation at https://github.com/Skyrat-SS13/Skyrat-tg/pull/14081

## Changelog
:cl:
config: Prisoners now scale their available positions based on how many Security positions are filled.
config: Scaling defaults to 2 prisoners per 1 security member if no config is supplied. Prisoners via scaling are capped at 12 if no config is supplied.
config: Prisoners are now rolled after all other jobs for this purpose.
/:cl:

### CODE REVIEW

**Comment by CandleJaxx** [comment]
goofball pr in 2022? based

**Comment by Iamgoofball** [comment]
gotta get that GBP up @CandleJaxx

File: code/controllers/configuration/entries/game_options.dm [hunk 1]
```diff
@@ -396,3 +396,13 @@
 	default = 1
 	min_val = 0
 	integer = FALSE
+
+// SKYRAT ADDITION BEGIN
+/datum/config_entry/number/prisoners_per_security // How many prisoners are spawned per security member? Set to 0 to disable this.
```
Comment (Zonespace27) [inline code/controllers/configuration/entries/game_options.dm]: modularize this

File: code/modules/jobs/job_types/_job.dm [hunk 2]
```diff
@@ -145,6 +150,12 @@
 	if(isnum(job_positions_edits["total_positions"]))
 		total_positions = job_positions_edits["total_positions"]
 
+//SKYRAT ADDITION BEGIN
+/datum/job/proc/skyrat_precheck()
```
Comment (Zonespace27) [inline code/modules/jobs/job_types/_job.dm]: ditto

File: code/modules/jobs/job_types/prisoner.dm [hunk 3]
```diff
@@ -25,6 +25,23 @@
 	rpg_title = "Defeated Miniboss"
 	job_flags = JOB_ANNOUNCE_ARRIVAL | JOB_CREW_MANIFEST | JOB_EQUIP_RANK | JOB_CREW_MEMBER | JOB_NEW_PLAYER_JOINABLE | JOB_ASSIGN_QUIRKS | JOB_CAN_BE_INTERN
 
+//SKYRAT ADDITION START
+/datum/job/prisoner/skyrat_precheck()
```
Comment (Zonespace27) [inline code/modules/jobs/job_types/prisoner.dm]: ditto

File: config/game_options.txt [hunk 4]
```diff
@@ -502,3 +502,11 @@ MAXFINE 2000
 
 ## Whether native FoV is enabled for all people.
 #NATIVE_FOV
+
+## SKYRAT ADDITION START
```
Comment (Zonespace27) [inline config/game_options.txt]: there is a skyrat config (config/skyrat/skyrat_config.txt)

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: config/game_options.txt (hunk 4 above)
Comment (Iamgoofball) [reply config/game_options.txt]: based

**Comment by Nerev4r** [comment]
This should probably be based first and foremost off COs and Wardens, as the rest of the Officers are generally going to have nothing to do with the prisoners.

**Comment by Iamgoofball** [comment]
@Nerev4r I considered that approach, but opted for general security population instead so that a shift with 7 security officers, a warden, and a HoS doesn't only have 2 prisoners given Skyrat's population. 

Unless things changed since I was banned, all of Security got involved if there was an incident at Perma so I think it'll be fine to allow Officers to count towards prisoners if we're scaling on the basis of "can security handle a riot on raw numbers".

File: modular_skyrat/modules/goofsec/code/prisoner_scaling_config.dm [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/datum/config_entry/number/prisoners_per_security // How many prisoners are spawned per security member? Set to 0 to disable this.
```
Comment (Zonespace27) [inline modular_skyrat/modules/goofsec/code/prisoner_scaling_config.dm:1]: actually tick the code file please

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/goofsec/code/prisoner_scaling_config.dm (hunk 5 above)
Comment (Iamgoofball) [reply modular_skyrat/modules/goofsec/code/prisoner_scaling_config.dm:1]: i got owned by VSC whoops

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by Higgin** [comment]
I think it should be 1:1 or weight COs more heavily relative to other sec roles, but this is a much more sensitive solution than mine and also reconfigurable.

**Comment by Gandalf2k15** [comment]
After testing this, I can for sure say that it's quite annoying having prisoner players ahelp to get a slot, and when we unlimit it, there generally isn't many issues with them now that they're following opfor. I think the flat out config change is better, but even then, 12 is not an issue! So I'll be closing this and the other PR.

**Comment by Iamgoofball** [comment]
yeet

**Pull request closed without merging**